chain/transaction.go

```diff
@@ -80,26 +79,7 @@ func (t *TransactionData) Sign(
 		return nil, err
 	}
 
-	signedTransaction := Transaction{
-		TransactionData: TransactionData{
-			Base:    t.Base,
-			Actions: t.Actions,
-		},
-		Auth: auth,
-	}
-
-	// Ensure transaction is fully initialized and correct by reloading it from
-	// bytes
-	size := len(msg) + consts.ByteLen + auth.Size()
-	p := codec.NewWriter(size, consts.NetworkSizeLimit)
-	if err := signedTransaction.Marshal(p); err != nil {
-		return nil, err
-	}
-	if err := p.Err(); err != nil {
-		return nil, err
-	}
-	p = codec.NewReader(p.Bytes(), consts.MaxInt)
-	return UnmarshalTx(p, actionCodec, authCodec)
+	return newTransaction(t.Base, t.Actions, auth)
```

Comment:
new, `Sign()` method uses `newTransaction()` to initialize a Transaction.
`actionCodec` and `authCodec` are now unnecessary.